pkg/console/operator/sync_v400.go

```diff
@@ -125,45 +118,43 @@ func (co *consoleOperator) sync_v400(updatedOperatorConfig *operatorv1.Console,
-	// or if we have not settled on the desired number of replicas or deployment is not up to date.
-	if toUpdate {
-		co.ConditionResourceSyncProgressing(updatedOperatorConfig, "Changes made during sync updates, additional sync expected.")
-	} else {
+	co.HandleProgressing(updatedOperatorConfig, "ResourceSyncUpdatesInProgress", func() error {
+		if toUpdate {
+			return errors.New("Changes made during sync updates, additional sync expected.")
+		}
 		version := os.Getenv("RELEASE_VERSION")
 		if !deploymentsub.IsAvailableAndUpdated(actualDeployment) {
-			co.ConditionResourceSyncProgressing(updatedOperatorConfig, fmt.Sprintf("Working toward version %s", version))
-		} else {
-			if co.versionGetter.GetVersions()["operator"] != version {
-				co.versionGetter.SetVersion("operator", version)
-			}
-			co.ConditionResourceSyncNotProgressing(updatedOperatorConfig)
+			return errors.New(fmt.Sprintf("Working toward version %s", version))
 		}
-	}
-
-	// the operand is available if all resources are:
-	// - present
-	// - if we have at least one ready replica
-	// - route is admitted
-	// available is currently defined as "met the users intent"
-	if !deploymentsub.IsReady(actualDeployment) {
-		msg := fmt.Sprintf("%v pods available for console deployment", actualDeployment.Status.ReadyReplicas)
-		klog.V(4).Infoln(msg)
-		co.ConditionDeploymentNotAvailable(updatedOperatorConfig, msg)
-	} else if !routesub.IsAdmitted(rt) {
-		klog.V(4).Infoln("console route is not admitted")
-		co.SetStatusCondition(
-			updatedOperatorConfig,
-			operatorv1.OperatorStatusTypeAvailable,
-			operatorv1.ConditionFalse,
-			"RouteNotAdmitted",
-			"console route is not admitted",
-		)
-	} else if actualDeployment.Status.Replicas == actualDeployment.Status.ReadyReplicas && actualDeployment.Status.Replicas == actualDeployment.Status.UpdatedReplicas {
-		co.ConditionDeploymentAvailable(updatedOperatorConfig, fmt.Sprintf("%v replicas ready at version %s", actualDeployment.Status.ReadyReplicas, os.Getenv("RELEASE_VERSION")))
-	} else {
-		co.ConditionDeploymentAvailable(updatedOperatorConfig, fmt.Sprintf("%v replicas ready", actualDeployment.Status.ReadyReplicas))
-	}
+		if co.versionGetter.GetVersions()["operator"] != version {
+			co.versionGetter.SetVersion("operator", version)
+		}
+		return nil
+	}())
+
+	co.HandleAvailable(updatedOperatorConfig, "DeploymentIsReady", func() error {
+		if !deploymentsub.IsReady(actualDeployment) {
+			msg := fmt.Sprintf("%v pods available for console deployment", actualDeployment.Status.ReadyReplicas)
+			klog.V(4).Infoln(msg)
+			return errors.New(msg)
+		}
+		return nil
+	}())
+	co.HandleAvailable(updatedOperatorConfig, "DeploymentIsUpdated", func() error {
+		if deploymentsub.IsReadyAndUpdated(actualDeployment) {
+			msg := fmt.Sprintf("%v replicas ready at version %s", actualDeployment.Status.ReadyReplicas, os.Getenv("RELEASE_VERSION"))
+			return errors.New(msg)
```

Comment:
would add also:
```
klog.V(4).Infoln(msg)
```

Reply:
+1
Def need to push these into a helper.